nixos/modules/services/mail/goeland.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.goeland;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  options.services.goeland = {
+    enable = mkEnableOption (mdDoc "goeland");
+
+    settings = mkOption {
+      description = mdDoc ''
+        Configuration of goeland.
+        See the [example config file](https://github.com/slurdge/goeland/blob/master/cmd/asset/config.default.toml) for the available options.
+      '';
+      default = { };
+      type = types.submodule {
+        freeformType = tomlFormat.type;
+      };
+    };
+    schedule = mkOption {
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+    databaseDirectory = mkOption {
+      type = types.path;
+      default = "/var/lib/goeland";
+      description = mdDoc "Directory where the goeland database will reside if using the `unseen` filter";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [ "d ${cfg.databaseDirectory} 0750 goeland goeland -" ];
```

Comment:
That would be good indeed! The only problem is that I remove the choice for the user to have the database somewhere else than /var/lib.
I don't mind imposing this to the user personally, but I don't know what's best when the program itself is not imposing any standard place. What do you think?

Reply:
I think it's reasonable to see if the use wants the defaults. If the databaseDirectory points to /var/lib, you let systemd manage it (and set the permissions). If the user sets it to a custom path, I'd expect the user to own the ownership decisions too (i.e. you don't try to chown it then): what if I want a custom group access for it or have it 0777 for whatever weird reason?

Reply:
Sounds good 👍🏽 
As you can see, I didn't put any logic in the code to only have `StateDirectory=` if it's the default value for databaseDiretory. I think the complexity isn't worth it but let me know if you think otherwise, I'll change it.

Reply:
It is a reasonable simplification, but I went through the [StateDirectory](https://sourcegraph.com/search?patternType=standard&q=context%3Aglobal+repo%3A%5Egithub%5C.com%2FNixOS%2Fnixpkgs%24+StateDirectory&sm=1&sq=repo%3A%5Egithub%5C.com%2FNixOS%2Fnixpkgs%24) usage in nixpkgs and it seems that the overwhelming majority of packages doesn't allow to configure it at all. Do you think that option would be useful for end users?

Reply:
Are you asking if we should make `StateDirectory` configurable, or are you asking if the existing `databaseDirectory` option is useful for the end user?
In the current state of the module, `StateDirectory` is not configurable. The `databaseDirectory` option only changes the config of goeland itself.
I don't think `StateDirectory` needs to be configurable.

Reply:
I’m saying we could remove databaseDitectory altogether, yeah.

Reply:
Well, I think having a default value for it does make things easier for the end user. If we remove the option but keep the default value, users will have to look through the code to find out a directory has been set for them by default.